src/xref2/component.ml

```diff
@@ -3,37 +3,69 @@ module Maps = Odoc_model.Paths.Identifier.Maps
 module ModuleMap = Map.Make (struct
   type t = Ident.module_
 
-  let compare a b = Ident.compare (a :> Ident.any) (b :> Ident.any)
+  let compare a b =
+    let i1 = match a with `LRoot (_, i) | `LModule (_, i) -> i in
+    let i2 = match b with `LRoot (_, i) | `LModule (_, i) -> i in
```

Comment:
Does this really have an effect ? I'm fine with specializing cases where the type has only one constructor (`TypeMap`) but I would be surprised if this leads to a speed up compared to the overhead of the `Map`.

Reply:
In my testing, `Ident.int_of_any` compiles to just an argument access (doing just two loads, ignoring the constructor) and is inlined inside `Ident.compare`, which then inline the integer comparison (which is not optimized, it does two `cmp`).
The new versions seems to compiles to the same thing but then calls the polymorphic compare. 

Do you have an input to benchmark this function ?

Reply:
What I was doing was turning off all the caches and doing `dune build @docgen` with `ODOC_BENCHMARK=true` (and with core_kernel changed to core if you've got v0.15)

Reply:
(https://github.com/jonludlam/odoc/commit/074aa6a50faf99efc5a8b15cd96288a3eadaf801, https://github.com/jonludlam/odoc/commit/0aa4240252eec24351156d05298cbdb3dde65250 and https://github.com/jonludlam/odoc/commit/ec5b94d8ffc118be2f5061322fac385e80375961)

Reply:
I have amended the commit so all that is left is the removal of a couple of constructors that weren't used.